src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -5101,3 +5102,47 @@ class CreateDatabaseScopedCredentialStatementSegment(BaseSegment):
             optional=True,
         ),
     )
+
+
+class CreateExternalDataSourceStatementSegment(BaseSegment):
+    """A statement to create an external data source.
+
+    https://learn.microsoft.com/en-us/sql/t-sql/statements/create-external-data-source-transact-sql?view=sql-server-ver16&tabs=dedicated#syntax
+    """
+
+    type = "create_external_data_source_statement"
+
+    match_grammar: Matchable = Sequence(
+        "CREATE",
+        "EXTERNAL",
+        "DATA",
+        "SOURCE",
+        Ref("ObjectReferenceSegment"),
+        "WITH",
+        Bracketed(
+            Delimited(
+                Sequence(
+                    Ref("TableLocationClause"),
+                    optional=True,
+                ),
+                Sequence(
+                    "CONNECTION_OPTIONS",
+                    Ref("EqualsSegment"),
+                    AnyNumberOf(Ref("QuotedLiteralSegmentOptWithN")),
+                    optional=True,
+                ),
+                Sequence(
+                    "CREDENTIAL",
+                    Ref("EqualsSegment"),
+                    Ref("ObjectReferenceSegment"),
+                    optional=True,
```

Comment:
You don't need optional=True in any of these, Delimited() extends OneOf(), so all of the *args you pass in are optional